server.go

```diff
@@ -647,6 +647,17 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 		quit:       make(chan struct{}),
 	}
 
+	// Start the low-level services once they are initialized.
+	//
+	// TODO(yy): break the server startup into four steps,
+	// 1. init the low-level services.
+	// 2. start the low-level services.
+	// 3. init the high-level services.
+	// 4. start the high-level services.
+	if err := s.startLowLevelServices(); err != nil {
```

Comment:
Nit: Would be great to have a description what's the basis of classification when a service is a low-level vs high-level service.

Reply:
yeah would love to - think we will identify them when we refactor.